hudi-client/src/main/java/org/apache/hudi/table/action/commit/MergeHelper.java

```diff
@@ -73,7 +74,11 @@
     } else {
       gReader = null;
       gWriter = null;
-      readSchema = upsertHandle.getWriterSchemaWithMetafields();
+      if (table.getConfig().updatePartialFields() && !StringUtils.isNullOrEmpty(table.getConfig().getLastSchema())) {
+        readSchema = new Schema.Parser().parse(table.getConfig().getLastSchema());
```

Comment:
similar comment as before. if we make config.getSchema() to always track full table schema, this can be simplified.